src/components/select/select-list/select-list.component.js

```diff
@@ -193,10 +230,30 @@ const SelectList = React.forwardRef(
       const indexOfMatch = getIndexOfMatch(highlightedValue);
 
       setCurrentOptionsListIndex(indexOfMatch);
-      updateListScrollTop(indexOfMatch, listRef.current);
-    }, [getIndexOfMatch, highlightedValue]);
+      updateListScrollTop(getScrollIndex(indexOfMatch), listRef.current);
+    }, [getIndexOfMatch, getScrollIndex, highlightedValue]);
 
     function getChildrenWithListProps() {
+      if (children.length && !children[0].props.value) {
+        const currentSelectedOption = childOptions[currentOptionsListIndex];
+        let currentOptionValue;
+
+        if (currentSelectedOption) {
+          currentOptionValue = currentSelectedOption.props.value;
+        }
+
+        return React.Children.map(children, (child) => {
+          if (!child.props.value) return child;
+
+          const newProps = {
+            onSelect: handleSelect,
+            isHighlighted: currentOptionValue === child.props.value,
+          };
+
+          return React.cloneElement(child, newProps);
+        });
+      }
+
```

Comment:
you don't have to create a separate list for children other than options, what you need is to use recurrence to check if the value under an index is an Option:
```
    const getNextHighlightableItemIndex = useCallback((key, indexOfHighlighted, lastIndex) => {
      let nextIndex = getNextIndexByKey(key, indexOfHighlighted, lastIndex);
      const nextElement = React.Children.toArray(children)[nextIndex];

      if (nextElement && nextElement.type !== Option) {
        nextIndex = getNextHighlightableItemIndex(key, nextIndex, lastIndex)
      }

      return nextIndex;
    }, [children]);
```
you only need to be sure that there is at least one Option to avoid infinite loop.

Doing that you could use the original index instead of relying on the option list.